js/controller.js

```diff
@@ -202,7 +202,7 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
         this.state.configLoaded = true;
         this.renderSkin();
 
-        var fullClass = (this.state.config.adScreen.showControlBar ? "" : " full");
+        var fullClass = (this.state.config.adScreen.showControlBar && this.state.showAdControls ? "" : " full");
```

Comment:
@chibzu, do we need to check this here? It looks like the below add/remove class takes care of everything.